### DISCUSSION TITLE
Set up and configure automated processing?

### DISCUSSION THREAD

**Discussion by JaredLodwick** [lllyasviel/Fooocus#2044] [Q&A] [answered]
Hello,

I'm currently working on a project that involves automated image processing, and I'm interested in integrating Fooocus into my workflow. I have a few questions that I hope you can help me with:

Automated Processing: Is there a way to use Fooocus for image processing without the browser-based UI? Specifically, I'm looking to automate the process of feeding images into Fooocus and retrieving the processed outputs programmatically.

Command Line/API Access: Does Fooocus support command-line operations or provide an API for remote triggering and control? This would be essential for setting up an automated pipeline.

Integration Details: If such functionality exists, could you provide some guidance or documentation on how to set it up? Any examples or references would be greatly appreciated.

Your software seems like a perfect fit for my project's requirements, and any assistance you could provide in this regard would be immensely helpful.

Thank you for your time and for developing Fooocus! It's a great tool and I'm really enjoying it!

**Comment by mashb1t** [comment, marked as answer]
Yes, Fooocus offers a Websocket based API OOTB (see footer) and https://github.com/konieshadow/Fooocus-API (fork) offers a REST API.
In the Footer of Fooocus you can find an example for each possible API call, but you can find more guidance here: https://github.com/lllyasviel/Fooocus/issues/1259

Hope this helps!

**Reply by JaredLodwick** [reply]
Oh look at that, totally missed it. Thanks so much!
